drv/ignition-api/src/lib.rs

```diff
@@ -144,39 +189,58 @@ impl From<Request> for u8 {
 #[derive(Copy, Clone, Debug, Default, PartialEq, Eq, AsBytes, FromBytes)]
 #[repr(C)]
 pub struct Counters {
-    status_received: u8,
-    hello_sent: u8,
-    request_sent: u8,
-    messages_dropped: u8,
+    pub status_received: u8,
+    pub hello_sent: u8,
+    pub request_sent: u8,
+    pub message_dropped: u8,
 }
 
 bitfield! {
     #[derive(Copy, Clone, Debug, Default, PartialEq, Eq, FromBytes, AsBytes)]
     #[repr(C)]
     pub struct LinkEvents(u8);
-    encoding_error, _: 0;
-    decoding_error, _: 1;
-    ordered_set_invalid, _: 2;
-    message_version_invalid, _: 3;
-    message_type_invalid, _: 4;
-    message_checksum_invalid, _: 5;
+    pub encoding_error, _: 0;
+    pub decoding_error, _: 1;
+    pub ordered_set_invalid, _: 2;
+    pub message_version_invalid, _: 3;
+    pub message_type_invalid, _: 4;
+    pub message_checksum_invalid, _: 5;
 }
 
 impl LinkEvents {
     pub const NONE: Self = Self(0b000000);
     pub const ALL: Self = Self(0b111111);
+
+    #[inline]
+    pub fn count(&self) -> usize {
+        self.0.count_ones().try_into().unwrap()
+    }
 }
 
 #[derive(
-    Copy, Clone, Debug, PartialEq, Eq, From, FromPrimitive, ToPrimitive, AsBytes,
+    Copy,
+    Clone,
+    Debug,
+    Display,
+    PartialEq,
+    Eq,
+    From,
+    FromPrimitive,
+    ToPrimitive,
+    AsBytes,
 )]
 #[repr(u8)]
-pub enum LinkSelect {
+pub enum TransceiverSelect {
```

Comment:
Can you add a comment explaining what this is?  I thought the main FPGA had a single link to each Ignition FPGA, but this is showing three separate links?